metadeploy/api/jobs.py

```diff
@@ -296,3 +261,31 @@ def expire_preflights():
 
 
 expire_preflights_job = job(expire_preflights)
+
+
+def create_scratch_org(*, plan_id, email, org_name):
+    plan = Plan.objects.get(id=plan_id)
+    repo_url = plan.version.product.repo_url
+    repo_owner, repo_name = extract_user_and_repo(repo_url)
+    commit_ish = plan.commit_ish
+    job_id = getattr(get_current_job(), "id")
+    try:
+        with local_github_checkout(
+            repo_owner, repo_name, commit_ish=commit_ish
+        ) as repo_root:
+            create_scratch_org_on_sf(
+                repo_owner=repo_owner,
+                repo_name=repo_name,
+                repo_url=repo_url,
+                repo_branch=commit_ish,
+                email=email,
+                project_path=repo_root,
+                org_name=org_name,
+            )
+    except Exception as e:
+        async_to_sync(notify_org_finished)(job_id, error=e)
+    finally:
```

Comment:
@wlonk In the error case, this triggers both success and error WS events.

```suggestion
    else:
```